brave/browser-laptop
# Fixes publisher not added to the ledger

Submitter Checklist:


Resolves #11553 - Publishers auto-included even with auto-include switch disabled
Resolves #11575 - Page data refactor
Resolves #11273 - Ledger table doesn't auto populate the publishers
Resolves #11274 - Site not added to publisher list even after spending ~10 mins on the site

Auditors:

Test Plan:

**for every test plan you need to clean your profile and enable payments**

Plan A
- open preferences, go to payments
- open a new tab
- load a site
- stay on that site for 15s
- switch to preference tab and site should be added

Plan B
- open preferences, go to payments
- open a new tab
- load a site
- stay on that site for 15s
- open a a new tab
- switch to preference tab and site should be added

Plan C
- do plan B
- switch to the site tab and stay on it for 15s
- open new tab and visit different site for 15s
- switch to preference tab and new site should be added and old site should have 2 visits

Plan D
- open preferences, go to payments
- change min visit time to 1min
- open new tab
- load a site stay on that site for 15s
- open another app (so that focus will be out)
- click back on opened site in brave and wait for 1m
- switch to preference tab and site should be added

Plan E
- do plan B
- go back to site tab and stay on that site for 15s
- switch to preference tab and site should have 1 visit, but 20+ seconds visit time

Plan F
- Open preferences, go to payments
- Disable `auto-include` settings 
- Visit some random sites so that the sites are added to the ledger table
- Ensure no site is auto included that gets added to the table

Plan G
- open preferences, go to payments
- open a new tab
- load a site stay on that site for 15s
- open a private tab
- load a new site stay on that site for 15s
- switch to preference tab and table should only contain site visited in normal tab and not from private tab

Plan H
- do plan A
- disable payments
- visit few more sites in new tabs
- switch to preference tab and ledger table should only have site from plan A


Reviewer Checklist:

Tests
